### PR TITLE
fix: modulepreload warnings

### LINKED ISSUE #6117: <link rel=modulepreload> has an invalid `as` value style
### Version

[v2.8.1](https://github.com/nuxt.js/releases/tag/v2.8.1)

### Reproduction link

[https://codesandbox.io/s/codesandbox-nuxt-z3zgt](https://codesandbox.io/s/codesandbox-nuxt-z3zgt)

### Steps to reproduce

Create `SPA` app, set `modern: true` and `extractCSS: true`
Check console warnings:

### What is expected ?

No warnings

### What is actually happening?

Warnings

### Additional comments?

`modulepreload` should be set to `scripts` only

<div align="right"><sub><em>This bug report is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/nuxt.js/issues/c9538">#c9538</a>)</em></sub></div>

### PR DESCRIPTION
**Pull request by dbachko**
## Types of changes

- [x] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description
Resolves #6117

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [ ] I have added tests to cover my changes (if not applicable, please state why)
- [x] All new and existing tests are passing.

### CODE REVIEW

File: packages/server/src/middleware/nuxt.js [hunk 1]
```diff
@@ -113,9 +113,9 @@ const defaultPushAssets = (preloadFiles, shouldPush, publicPath, options) => {
 
     const { crossorigin } = options.build
     const cors = `${crossorigin ? ` crossorigin=${crossorigin};` : ''}`
-    const ref = modern ? 'modulepreload' : 'preload'
+    const rel = `${(modern && asType === 'script') ? 'module' : ''}preload`
```
Comment (clarkdo) [inline packages/server/src/middleware/nuxt.js]:
```suggestion
    const rel = modern && asType === 'script' ? 'modulepreload' : 'preload'
```

Comment (clarkdo) [reply packages/server/src/middleware/nuxt.js]: BTW, could you add a comment here (describe that modulepreload only supports script-like `as` value) for others can have better understanding  ?
https://html.spec.whatwg.org/multipage/links.html#link-type-modulepreload

**Review by clarkdo — COMMENTED** [review]
Really good catch, thank you so much

File: packages/server/src/middleware/nuxt.js (hunk 1 above)
Comment (dbachko) [reply packages/server/src/middleware/nuxt.js]: Done

File: packages/vue-renderer/src/renderers/spa.js [hunk 2]
```diff
@@ -109,7 +109,8 @@ export default class SPARenderer extends BaseRenderer {
             if (asType === 'font') {
               extra = ` type="font/${extension}"${cors ? '' : ' crossorigin'}`
             }
-            return `<link rel="${modern ? 'module' : ''}preload"${cors} href="${publicPath}${file}"${
+            const rel = `${(modern && asType === 'script') ? 'module' : ''}preload`
```
Comment (clarkdo) [inline packages/vue-renderer/src/renderers/spa.js]: Could you please apply same change as above one ? 😄

**Review by clarkdo — APPROVED** [review]

**Comment by clarkdo** [comment]
@dbachko Thanks!

**Pull request merged**